packages/share_plus/share_plus/lib/src/share_plus_web.dart

```diff
@@ -156,10 +156,20 @@ class SharePlusWebPlugin extends SharePlatform {
     String? subject,
     String? text,
     Rect? sharePositionOrigin,
+    List<String>? fileNameOverrides,
   }) async {
+    assert(
+        fileNameOverrides == null || files.length == fileNameOverrides.length);
     final webFiles = <web.File>[];
-    for (final xFile in files) {
-      webFiles.add(await _fromXFile(xFile));
+    if (fileNameOverrides != null) {
+      for (final entry in files.asMap().entries) {
+        webFiles.add(await _fromXFile(entry.value,
+            nameOverride: fileNameOverrides[entry.key]));
+      }
+    } else {
+      for (final xFile in files) {
+        webFiles.add(await _fromXFile(xFile));
+      }
```

Comment:
There is probably a simpler way to write this with just a for loop with an index

Something like this (haven't checked if correct)

```dart
for (var index = 0; index < files.length; index++) {
  xFile = files[index];
  filename = fileNameOverrides?.elementAt(index);
   webFiles.add(await _fromXFile(xFile, filename));
}
```

Reply:
You're right, I have refactored that part to use a single for loop, like you suggested